test/utils/testconfig/testconfig.go

```diff
@@ -0,0 +1,295 @@
+package testconfig
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"context"
+
+	ptpv1 "github.com/openshift/ptp-operator/api/v1"
+	testclient "github.com/openshift/ptp-operator/test/utils/client"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ConfigCheck                  = "CONFIGCHECK"
+	OrdinaryClockString          = "OC"
+	BoundaryClockString          = "BC"
+	DualNICBoundaryClockString   = "DualNICBC"
+	DiscoveryString              = "Discovery"
+	legacyDiscoveryString        = "true"
+	NoneString                   = ""
+	StartString                  = ""
+	InitStatusString             = "init"
+	ConfiguredStatusString       = "configured"
+	DiscoverySuccessStatusString = "discoverySuccess"
+	DiscoveryFailureStatusString = "discoveryFailure"
+)
+
+type ConfigStatus int64
+
+const (
+	Start ConfigStatus = iota
+	InitStatus
+	ConfiguredStatus
+	DiscoverySuccessStatus
+	DiscoveryFailureStatus
+)
+
+type PTPMode int64
+
+const (
+	OrdinaryClock PTPMode = iota
+	BoundaryClock
+	DualNICBoundaryClock
+	Discovery
+	None
+)
+
+type TestConfig struct {
+	PtpModeDesired    PTPMode
+	PtpModeDiscovered PTPMode
+	Status            ConfigStatus
+	DiscoveredSlavePtpConfig,
+	DiscoveredSlavePtpConfigSecondary,
+	DiscoveredMasterPtpConfig ptpDiscoveryRes
+}
+type ptpDiscoveryRes struct {
+	Label  string
+	Config ptpv1.PtpConfig
+}
+
+func (obj ptpDiscoveryRes) String() string {
+	return fmt.Sprintf("[label=%s, config name=%s]", obj.Label, obj.Config.Name)
+}
+func (obj TestConfig) String() string {
+	return fmt.Sprintf("PtpModeDesired=%s, PtpModeDiscovered=%s, Status=%s, DiscoveredSlavePtpConfig=%s, DiscoveredSlavePtpConfigSecondary=%s, DiscoveredMasterPtpConfig=%s", obj.PtpModeDesired, obj.PtpModeDiscovered, obj.Status, obj.DiscoveredSlavePtpConfig, obj.DiscoveredSlavePtpConfigSecondary, obj.DiscoveredMasterPtpConfig)
+}
+
+func (status ConfigStatus) String() string {
+	switch status {
+	case Start:
+		return StartString
+	case InitStatus:
+		return InitStatusString
+	case ConfiguredStatus:
+		return ConfiguredStatusString
+	case DiscoverySuccessStatus:
+		return DiscoverySuccessStatusString
+	case DiscoveryFailureStatus:
+		return DiscoveryFailureStatusString
+	default:
+		return StartString
+	}
+}
+
+func (mode PTPMode) String() string {
+	switch mode {
+	case OrdinaryClock:
+		return OrdinaryClockString
+	case BoundaryClock:
+		return BoundaryClockString
+	case DualNICBoundaryClock:
+		return DualNICBoundaryClockString
+	case Discovery:
+		return DiscoveryString
+	case None:
```

Comment:
Why None mode is required. What is None means,? If None is set then it should set to default which is an ordinary clock , right ?

Reply:
right, none is just to support the case when the env variable is ""